packages/flutter/lib/src/material/flat_button.dart

```diff
@@ -138,7 +141,74 @@ class FlatButton extends MaterialButton {
     required Widget child,
     double? height,
     double? minWidth,
-  }) : assert(clipBehavior != null),
+  }) : _buttonType = _FlatButtonType.material,
+       assert(clipBehavior != null),
+       assert(autofocus != null),
+       super(
+         key: key,
+         height: height,
+         minWidth: minWidth,
+         onPressed: onPressed,
+         onLongPress: onLongPress,
+         onHighlightChanged: onHighlightChanged,
+         mouseCursor: mouseCursor,
+         textTheme: textTheme,
+         textColor: textColor,
+         disabledTextColor: disabledTextColor,
+         color: color,
+         disabledColor: disabledColor,
+         focusColor: focusColor,
+         hoverColor: hoverColor,
+         highlightColor: highlightColor,
+         splashColor: splashColor,
+         colorBrightness: colorBrightness,
+         padding: padding,
+         visualDensity: visualDensity,
+         shape: shape,
+         clipBehavior: clipBehavior,
+         focusNode: focusNode,
+         autofocus: autofocus,
+         materialTapTargetSize: materialTapTargetSize,
+         child: child,
+       );
+
+  /// Create a simple text button that displays [CupertinoButton] in iOS.
+  ///
+  /// The [autofocus] and [clipBehavior] arguments must not be null.
+  ///
+  /// The [onLongPress], [onHighlightChanged], [mouseCursor], [textTheme],
+  /// [textColor], [disabledTextColor], [focusColor], [hoverColor],
+  /// [highlightColor], [splashColor], [colorBrightness], [visualDensity],
+  /// [shape], [clipBehavior], [focusNode], [autofocus], [materialTapTargetSize]
+  /// and [height], will be ignored in iOS.
```

Comment:
We don't have to ignore height if we wrapped CupertinoButton with a ConstrainedBox with both width and height. That's what the material button does with it anyway https://github.com/flutter/flutter/blob/13896b3bd1858687a06b1639d60c877206170df8/packages/flutter/lib/src/material/material_button.dart#L424.